cloud/scope/powervs_cluster.go

```diff
@@ -1744,24 +1753,29 @@ func (s *PowerVSClusterScope) fetchResourceGroupID() (string, error) {
 	return "", err
 }
 
-// getVPCRegion returns region associated with VPC zone.
-func (s *PowerVSClusterScope) getVPCRegion() *string {
-	if s.IBMPowerVSCluster.Spec.VPC != nil {
-		return s.IBMPowerVSCluster.Spec.VPC.Region
-	}
-	// if vpc region is not set try to fetch corresponding region from power vs zone
+// getTransitGatewayLocationAndRouting returns appropriate location and routing suitable for transit gateway.
+// routing indicates whether to enable global routing on transit gateway or not.
+// returns true when powervs and vpc region are not common otherwise false.
+func (s *PowerVSClusterScope) getTransitGatewayLocationAndRouting() (*string, *bool) {
 	zone := s.Zone()
 	if zone == nil {
 		s.Info("powervs zone is not set")
-		return nil
+		return nil, nil
 	}
 	region := endpoints.ConstructRegionFromZone(*zone)
-	vpcRegion, err := genUtil.VPCRegionForPowerVSRegion(region)
+
+	if s.IBMPowerVSCluster.Spec.VPC != nil {
+		routing := !genUtil.IsCommonPowerVSAndVPCRegion(region, *s.IBMPowerVSCluster.Spec.VPC.Region)
```

Comment:
Refactored the func a bit, I would like to keep the var as globalRouting since we are going to assign it to `CreateTransitGatewayOptions.Global` field.

Reply:
that makes sense.